### PR TITLE
[IMP] developer/orm: add read_group relative granularity

### PR DESCRIPTION
**Pull request by VincentSchippefilt**
references https://github.com/odoo/odoo/pull/159528 Task: 3836802

### CODE REVIEW

File: content/developer/reference/backend/orm.rst [hunk 1]
```diff
@@ -885,6 +885,9 @@ A domain is a list of criteria, each criterion being a triple (either a
     a field name of the current model, or a relationship traversal through
     a :class:`~odoo.fields.Many2one` using dot-notation e.g. ``'street'``
     or ``'partner_id.country'``
+    If the field is a date field, you can also specify a part of the date using ``'field_name.granularity'``.
+    The supported granularities are ``'quarter'``, ``'month'``, ``'week'``.
+    They all use a integeger as value.
```
Comment (pro-odoo) [inline content/developer/reference/backend/orm.rst:890]: integer

File: content/developer/reference/backend/orm.rst [hunk 2]
```diff
@@ -885,6 +885,9 @@ A domain is a list of criteria, each criterion being a triple (either a
     a field name of the current model, or a relationship traversal through
     a :class:`~odoo.fields.Many2one` using dot-notation e.g. ``'street'``
     or ``'partner_id.country'``
+    If the field is a date field, you can also specify a part of the date using ``'field_name.granularity'``.
```
Comment (Feyensv) [inline content/developer/reference/backend/orm.rst:888]:
```suggestion
    If the field is a date(time) field, you can also specify a part of the date using ``'field_name.granularity'``.
```

File: content/developer/reference/backend/orm.rst [hunk 3]
```diff
@@ -885,6 +885,9 @@ A domain is a list of criteria, each criterion being a triple (either a
...
+    The supported granularities are ``'quarter_number'``, ``'month_number'``, ``'week_number'``.
+    They all use a integer as value.
```
Comment (AntoineVDV) [inline content/developer/reference/backend/orm.rst:890]:
```suggestion
    or ``'partner_id.country'``
    If the field is a date field, you can also specify a part of the date using ``'field_name.granularity'``.
    The supported granularities are ``'quarter_number'``, ``'month_number'``, ``'week_number'``.
    They all use an integer as value.
```

File: content/developer/reference/backend/orm.rst [hunk 4]
```diff
@@ -981,6 +984,11 @@ Domain criteria can be combined using logical operators in *prefix* form:
         [('invoice_status', '=', 'to invoice'),
          ('order_line', 'any', [('product_id.qty_available', '<=', 0)])]
 
+
+    To search for all partners that are born in the month February::
+
+        [('birthday.month_number','=',2)]
```
Comment (Feyensv) [inline content/developer/reference/backend/orm.rst:990]:
```suggestion
        [('birthday.month_number', '=', 2)]
```
nitpicking, but domains are clearer when there are spaces between tuple parts.

File: content/developer/reference/backend/orm.rst [hunk 5]
```diff
@@ -981,6 +984,11 @@ Domain criteria can be combined using logical operators in *prefix* form:
         [('invoice_status', '=', 'to invoice'),
          ('order_line', 'any', [('product_id.qty_available', '<=', 0)])]
 
+
+    To search for all partners that are born in the month February::
```
Comment (AntoineVDV) [inline content/developer/reference/backend/orm.rst:988]:
```suggestion
    To search for all partners born in the month of February::
```

File: content/developer/reference/backend/orm.rst (hunk 4 above)
Comment (AntoineVDV) [inline content/developer/reference/backend/orm.rst:990]:
```suggestion
        [('birthday.month_number', '=', 2)]
```

File: content/developer/reference/backend/orm.rst (hunk 3 above)
Comment (Feyensv) [inline content/developer/reference/backend/orm.rst:890]:
```suggestion
    They all use an integer as value.
```

File: content/developer/reference/backend/orm/changelog.rst [hunk 6]
```diff
@@ -4,6 +4,12 @@
 Changelog
 =========
 
+Odoo Online version 17.3
+========================
+
+- in read_group, _read_group and in domains, we now support grouping by date parts : month_number, quarter_number and week_number: `#159528 <https://github.com/odoo/odoo/pull/159528>`_.
```
Comment (AntoineVDV) [inline content/developer/reference/backend/orm/changelog.rst:10]:
```suggestion
- We can now group by date parts (`month_number`, `quarter_number` and `week_number`) in `read_group`, `_read_group` and domains with `#159528 <https://github.com/odoo/odoo/pull/159528>`_.
```

**Review by AntoineVDV — COMMENTED** [review]
@robodoo delegate+

**Comment by AntoineVDV** [comment]
@robodoo squash

**Review by Feyensv — APPROVED** [review]
Very minor comments.
Please be aware that there is no concept of odoo modules here, so the commit (and PR) target approximative scopes in the documentation (`developer/orm` in this case).

@robodoo delegate+

Edit: already resolved some comments already highlighted in Antoine review :D

**Comment by VincentSchippefilt** [comment]
robodoo r+

**Comment by VincentSchippefilt** [comment]
robodoo r+

**Pull request closed without merging**